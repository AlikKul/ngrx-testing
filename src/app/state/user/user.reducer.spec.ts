import { User } from '@state/user/user.model';
import {
  AddUser,
  AddUserFail,
  AddUserSuccess,
  LoadUser,
  LoadUserFail,
  LoadUserSuccess,
  LoadUsers,
  LoadUsersFail,
  LoadUsersSuccess,
  SelectUser,
  UpdateUser,
  UpdateUserFail,
  UpdateUserSuccess,
  UpdateUsers,
  UpdateUsersSuccess
} from './user.actions';
import * as fromUser from './user.reducer';

describe('User Reducer', () => {
  let user: User = {
    id: 1,
    firstName: 'Anakin',
    lastName: 'Skywalker'
  };

  describe('undefined action', () => {
    it('should return the default state', () => {
      const action = {} as any;
      const result = fromUser.reducer(undefined, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Add User', () => {
    it('should toggle loading state', () => {
      const action = new AddUser({ user });
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Add User Success', () => {
    it('should add a user to state', () => {
      const action = new AddUserSuccess({ user });
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Add User Fail', () => {
    it('should update error to state', () => {
      const action = new AddUserFail({ error: new Error() });
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Load User', () => {
    it('should toggle loading state', () => {
      const action = new LoadUser({ id: user.id });
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Load User Success', () => {
    it('should load a user to state', () => {
      const action = new LoadUserSuccess({ user });
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Load User Fail', () => {
    it('should update error in state', () => {
      const action = new LoadUserFail({ error: new Error() });
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Load Users', () => {
    it('should load a user to state', () => {
      const action = new LoadUsers();
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Load Users Success', () => {
    it('should add all users to state', () => {
      const users = [user];
      const action = new LoadUsersSuccess({ users });
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Load Users Fail', () => {
    it('should update error in state', () => {
      const users = [user];
      const action = new LoadUsersFail({ error: new Error() });
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Update User', () => {
    it('should load a user to state', () => {
      const action = new UpdateUser({
        user: { ...user, firstName: 'Darth', lastName: 'Vader' }
      });
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Update User Success', () => {
    it('should add all users to state', () => {
      const users = [user];
      const action = new UpdateUserSuccess({
        update: {
          id: user.id,
          changes: {
            ...user,
            firstName: 'Darth',
            lastName: 'Vader'
          }
        }
      });
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Update User Fail', () => {
    it('should update error in state', () => {
      const users = [user];
      const action = new UpdateUserFail({ error: new Error() });
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Update Users', () => {
    it('should load a user to state', () => {
      const action = new UpdateUsers({
        users: [{ ...user, firstName: 'Darth', lastName: 'Vader' }]
      });
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Update Users Success', () => {
    it('should add all users to state', () => {
      const users = [user];
      const action = new UpdateUsersSuccess({
        update: [
          {
            id: user.id,
            changes: {
              ...user,
              firstName: 'Darth',
              lastName: 'Vader'
            }
          }
        ]
      });
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[User] Select User', () => {
    it('should set the selectedUserId property in state', () => {
      const action = new SelectUser({ id: user.id });
      const result = fromUser.reducer(fromUser.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });
});
