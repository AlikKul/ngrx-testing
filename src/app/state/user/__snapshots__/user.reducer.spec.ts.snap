// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Reducer [User] Add User Fail should update error to state 1`] = `
Object {
  "entities": Object {},
  "error": [Error],
  "ids": Array [],
  "loading": false,
  "selectedUserId": null,
}
`;

exports[`User Reducer [User] Add User Success should add a user to state 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "firstName": "Anakin",
      "id": 1,
      "lastName": "Skywalker",
    },
  },
  "ids": Array [
    1,
  ],
  "loading": false,
  "selectedUserId": null,
}
`;

exports[`User Reducer [User] Add User should toggle loading state 1`] = `
Object {
  "entities": Object {},
  "error": undefined,
  "ids": Array [],
  "loading": true,
  "selectedUserId": null,
}
`;

exports[`User Reducer [User] Load User Fail should update error in state 1`] = `
Object {
  "entities": Object {},
  "error": [Error],
  "ids": Array [],
  "loading": false,
  "selectedUserId": null,
}
`;

exports[`User Reducer [User] Load User Success should load a user to state 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "firstName": "Anakin",
      "id": 1,
      "lastName": "Skywalker",
    },
  },
  "ids": Array [
    1,
  ],
  "loading": false,
  "selectedUserId": null,
}
`;

exports[`User Reducer [User] Load User should toggle loading state 1`] = `
Object {
  "entities": Object {},
  "error": undefined,
  "ids": Array [],
  "loading": true,
  "selectedUserId": null,
}
`;

exports[`User Reducer [User] Load Users Fail should update error in state 1`] = `
Object {
  "entities": Object {},
  "error": [Error],
  "ids": Array [],
  "loading": false,
  "selectedUserId": null,
}
`;

exports[`User Reducer [User] Load Users Success should add all users to state 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "firstName": "Anakin",
      "id": 1,
      "lastName": "Skywalker",
    },
  },
  "ids": Array [
    1,
  ],
  "loading": false,
  "selectedUserId": null,
}
`;

exports[`User Reducer [User] Load Users should load a user to state 1`] = `
Object {
  "entities": Object {},
  "error": undefined,
  "ids": Array [],
  "loading": true,
  "selectedUserId": null,
}
`;

exports[`User Reducer [User] Select User should set the selectedUserId property in state 1`] = `
Object {
  "entities": Object {},
  "ids": Array [],
  "loading": false,
  "selectedUserId": 1,
}
`;

exports[`User Reducer [User] Update User Fail should update error in state 1`] = `
Object {
  "entities": Object {},
  "error": [Error],
  "ids": Array [],
  "loading": false,
  "selectedUserId": null,
}
`;

exports[`User Reducer [User] Update User Success should add all users to state 1`] = `
Object {
  "entities": Object {},
  "ids": Array [],
  "loading": false,
  "selectedUserId": null,
}
`;

exports[`User Reducer [User] Update User should load a user to state 1`] = `
Object {
  "entities": Object {},
  "error": undefined,
  "ids": Array [],
  "loading": true,
  "selectedUserId": null,
}
`;

exports[`User Reducer [User] Update Users Success should add all users to state 1`] = `
Object {
  "entities": Object {},
  "ids": Array [],
  "loading": false,
  "selectedUserId": null,
}
`;

exports[`User Reducer [User] Update Users should load a user to state 1`] = `
Object {
  "entities": Object {},
  "error": undefined,
  "ids": Array [],
  "loading": true,
  "selectedUserId": null,
}
`;

exports[`User Reducer undefined action should return the default state 1`] = `
Object {
  "entities": Object {},
  "ids": Array [],
  "loading": false,
  "selectedUserId": null,
}
`;
